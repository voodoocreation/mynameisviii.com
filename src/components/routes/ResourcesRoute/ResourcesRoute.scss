@import "../../../scss/variables", "../../../scss/mixins";

.ResourcesRoute--listings {
  @include panel-1();

  margin-bottom: $spacing;
  padding: $spacing;

  h2 {
    @include line-before-after();

    margin-top: 0;
  }

  &--items {
    display: flex;
    flex-wrap: wrap;
    margin: (-$spacing-half);
  }
}
