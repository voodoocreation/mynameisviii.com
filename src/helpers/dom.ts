import getScrollbarWidth from "scrollbar-size";

if (typeof window !== "undefined") {
  window.scrollState = {
    count: 0,
    left: 0,
    top: 0
  };
}

export const isServer = () => typeof window === "undefined" || window.isServer;

export const isInViewport = (element?: HTMLElement | null) => {
  if (!element) {
    return false;
  }

  const bounds = element.getBoundingClientRect();
  const viewport = {
    height: window.innerHeight || document.documentElement.clientHeight,
    width: window.innerWidth || document.documentElement.clientWidth
  };

  return (
    bounds.top >= 0 &&
    bounds.right <= viewport.width &&
    bounds.bottom <= viewport.height &&
    bounds.left >= 0
  );
};

export const isAlmostInViewport = (
  element?: HTMLElement | null,
  offset = 0
) => {
  if (!element) {
    return false;
  }

  const bounds = element.getBoundingClientRect();
  const viewport = {
    height: window.innerHeight || document.documentElement.clientHeight,
    width: window.innerWidth || document.documentElement.clientWidth
  };

  return (
    viewport.height - bounds.top > -offset &&
    bounds.right > -offset &&
    bounds.bottom > -offset &&
    viewport.width - bounds.left > -offset
  );
};

export const lockScroll = () => {
  const { body } = document;
  const html = document.documentElement;

  window.scrollState.count += 1;

  if (!html.classList.contains("isLocked")) {
    const existingPadding = parseInt(
      getComputedStyle(body, null).getPropertyValue("padding-right"),
      10
    );
    const scrollbarWidth = getScrollbarWidth();
    const newPadding =
      isNaN(existingPadding) || !isFinite(existingPadding)
        ? scrollbarWidth
        : scrollbarWidth + existingPadding;
    const scrollElement =
      body.scrollTop > 0 || html.scrollTop === 0 ? body : html;

    window.scrollState.top = scrollElement.scrollTop;
    window.scrollState.left = scrollElement.scrollLeft;

    html.classList.add("isLocked");
    body.style.paddingRight = `${newPadding}px`;
  }
};

export const unlockScroll = () => {
  const { body } = document;
  const html = document.documentElement;

  if (window.scrollState.count <= 1 && html.classList.contains("isLocked")) {
    body.style.paddingRight = "";
    html.classList.remove("isLocked");
    window.scrollTo(window.scrollState.left, window.scrollState.top);
  }

  if (window.scrollState.count > 0) {
    window.scrollState.count -= 1;
  }
};
