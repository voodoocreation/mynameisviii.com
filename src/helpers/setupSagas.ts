import merge from "lodash.merge";
import SagaTester from "redux-saga-tester";

import rootReducer from "../reducers/root.reducers";
import rootSaga from "../sagas/root.sagas";

export default (fromTestStore: {}, fromTestPorts: {}) => {
  const initialState = merge({}, fromTestStore);
  const ports = merge(
    {
      api: {},
      dataLayer: [],
      maps: {}
    },
    fromTestPorts
  );

  const sagaTester = new SagaTester({
    initialState,
    reducers: rootReducer
  });
  sagaTester.start(rootSaga(ports));

  return {
    dispatch: (action: any) => sagaTester.dispatch(action),
    findAction: (actionFromTest: any) =>
      sagaTester
        .getCalledActions()
        .find((action: any) => action.type === actionFromTest().type),
    sagaTester,
    store: () => sagaTester.getState()
  };
};
