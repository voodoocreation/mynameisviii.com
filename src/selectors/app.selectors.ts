import { TStoreState } from "../reducers/root.reducers";

export const getCurrentRoute = (state: TStoreState) => state.app.currentRoute;

export const getAppError = (state: TStoreState) => state.app.error;

export const isAppLoading = (state: TStoreState) => state.app.isLoading;

export const hasNewVersion = (state: TStoreState) => state.app.hasNewVersion;

export const isNavOpen = (state: TStoreState) => state.app.isNavOpen;

export const isOnline = (state: TStoreState) => state.app.isOnline;
