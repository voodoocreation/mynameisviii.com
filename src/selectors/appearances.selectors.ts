import { createSelector, defaultMemoize } from "reselect";

import { TStoreState } from "../reducers/root.reducers";

export const hasAppearancesError = (state: TStoreState) =>
  state.appearances.hasError;

export const getAppearances = defaultMemoize(
  (state: TStoreState) => state.appearances.items
);

export const getAppearancesCount = createSelector(
  getAppearances,
  (appearances) => Object.keys(appearances).length
);

export const getAppearancesAsArray = createSelector(
  getAppearances,
  (appearances) => Object.values(appearances)
);

export const getUpcomingAppearances = createSelector(
  getAppearancesAsArray,
  (appearances) =>
    appearances
      .filter((appearance) => appearance.finishingAt > new Date().toISOString())
      .sort(
        (a, b) =>
          new Date(a.startingAt).getTime() - new Date(b.startingAt).getTime()
      )
);

export const getPastAppearances = createSelector(
  getAppearancesAsArray,
  (appearances) =>
    appearances
      .filter(
        (appearance) => appearance.finishingAt <= new Date().toISOString()
      )
      .sort(
        (a, b) =>
          new Date(b.startingAt).getTime() - new Date(a.startingAt).getTime()
      )
);

export const getCurrentAppearanceSlug = (state: TStoreState) =>
  state.appearances.currentSlug;

export const getCurrentAppearance = createSelector(
  getAppearances,
  getCurrentAppearanceSlug,
  (appearances, slug) => (slug ? appearances[slug] : undefined)
);

export const getAppearancesLastEvaluatedKey = (state: TStoreState) =>
  state.appearances.lastEvaluatedKey;

export const getAppearancesLastEvaluatedKeyAsString = createSelector(
  getAppearancesLastEvaluatedKey,
  (lastEvaluatedKey) =>
    !lastEvaluatedKey
      ? undefined
      : btoa(
          encodeURIComponent(
            JSON.stringify({
              IsActive: lastEvaluatedKey.isActive,
              Slug: lastEvaluatedKey.slug,
              StartingAt: lastEvaluatedKey.startingAt,
            })
          )
        )
);

export const getHasAllAppearances = (state: TStoreState) =>
  state.appearances.hasAllItems;

export const getAppearancesIsLoading = (state: TStoreState) =>
  state.appearances.isLoading;

export const getCurrentAppearanceLocation = (state: TStoreState) =>
  state.appearances.currentLocation;
