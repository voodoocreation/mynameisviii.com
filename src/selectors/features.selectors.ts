import { TStoreState } from "../reducers/root.reducers";

export const getFeatures = (state: TStoreState) => state.features.items;

export const hasFeature = (state: TStoreState, feature: string) =>
  getFeatures(state).includes(feature);
