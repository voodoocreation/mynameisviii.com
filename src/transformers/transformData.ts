export const arrayToAssoc = (
  array: Array<{}>,
  key: string,
  failSilently = true
) =>
  array.reduce(
    (
      acc: { [index: string]: {} },
      curr: { [index: string]: any },
      index: number
    ) => {
      if (curr[key] !== undefined) {
        acc[curr[key]] = curr;
      } else if (!failSilently) {
        throw new Error(`Key '${key}' not found at index ${index}`);
      }
      return acc;
    },
    {}
  );

export const assocToArray: (object: any) => any = object =>
  Object.keys(object).reduce((acc: Array<{}>, curr: string) => {
    acc.push(object[curr]);
    return acc;
  }, []);

export const tryParseJson = (json: string | Error) => {
  let result;

  try {
    result = json.toString ? JSON.parse(json.toString()) : json;
  } catch (error) {
    result = json;
  }

  return result;
};

export const lengthToDuration = (length: string) => {
  const segments = length.split(":");

  switch (segments.length) {
    default:
    case 2:
      return `PT${segments[0]}M${segments[1]}S`;

    case 3:
      return `PT${segments[0]}H${segments[1]}M${segments[2]}S`;
  }
};

export const absUrl = (path: string) =>
  process.env.NODE_ENV !== "production"
    ? `http://localhost:${process.env.PORT || window.location.port}${path}`
    : `http://${
        typeof window === "undefined"
          ? process.env.DOMAIN
          : window.location.host
      }${path}`;

export const extractDomain = (url: string) => url.split("/")[2];
