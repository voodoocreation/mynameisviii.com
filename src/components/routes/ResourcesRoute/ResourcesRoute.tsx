import cn from "classnames";
import Head from "next/head";
import * as React from "react";
import { FormattedMessage, InjectedIntl } from "react-intl";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";
import { ActionCreator } from "typescript-fsa";

import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";
import { absUrl } from "../../../transformers/transformData";
import OfflineNotice from "../../connected/OfflineNotice/OfflineNotice";
import ButtonBar from "../../presentation/ButtonBar/ButtonBar";
import LoadButton from "../../presentation/LoadButton/LoadButton";
import NoResults from "../../presentation/NoResults/NoResults";
import PageHeader from "../../presentation/PageHeader/PageHeader";
import ResourceListing from "../../presentation/ResourceListing/ResourceListing";

import * as actions from "../../../actions/root.actions";
import * as selectors from "../../../selectors/root.selectors";

interface IStoreProps {
  error?: IError;
  hasAllResources: boolean;
  isLoading: boolean;
  resources: {
    [index: string]: IResource[];
  };
  resourcesCount: number;
}

interface IDispatchProps {
  fetchMoreResources: ActionCreator<{}>;
  fetchResources: ActionCreator<{}>;
}

interface IProps extends IStoreProps, IDispatchProps {
  intl: InjectedIntl;
}

interface IState {
  loadedListings: {
    [index: string]: boolean;
  };
}

class ResourcesRoute extends React.Component<IProps, IState> {
  public static async getInitialProps(props: any) {
    const { isServer, store } = props.ctx;

    const state = store.getState();

    if (isServer || selectors.getResourcesCount(state) === 0) {
      store.dispatch(actions.fetchResources.started({}));
    } else if (!selectors.getHasAllResources(state)) {
      store.dispatch(actions.fetchMoreResources.started({}));
    }
  }

  public readonly state = {
    loadedListings: {}
  };

  public render() {
    const { error, hasAllResources, resources, resourcesCount } = this.props;
    const { formatMessage } = this.props.intl;

    const hasLoadedAllListings =
      resourcesCount === Object.keys(this.state.loadedListings).length;

    const isLoading = this.props.isLoading || !hasLoadedAllListings;

    const loadMoreButton = hasAllResources ? null : (
      <LoadButton
        isLoading={isLoading}
        isScrollLoadEnabled={!error}
        onLoad={this.onLoadMore}
      >
        {!error ? (
          <FormattedMessage id="LOAD_MORE" />
        ) : (
          <FormattedMessage id="TRY_AGAIN" />
        )}
      </LoadButton>
    );

    const pageTitle = formatMessage({ id: "RESOURCES_TITLE" });
    const pageDescription = formatMessage({ id: "RESOURCES_DESCRIPTION" });

    return (
      <article className={cn("ResourcesRoute", { hasLoadedAllListings })}>
        <Head>
          <title>{pageTitle}</title>

          <meta content={pageDescription} name="description" />

          <meta property="og:title" content={pageTitle} />
          <meta property="og:description" content={pageDescription} />
          <meta property="og:url" content={absUrl("/resources")} />
          <meta property="og:type" content="website" />
          <meta
            property="og:image"
            content="https://s3.amazonaws.com/mynameisviii-static/resources-og.jpg"
          />
          <meta property="og:image:width" content="1200" />
          <meta property="og:image:height" content="630" />
        </Head>

        <PageHeader>
          <FormattedMessage id="RESOURCES_TITLE" />
        </PageHeader>

        {error ? <OfflineNotice /> : null}

        {resourcesCount > 0
          ? Object.keys(resources).map(this.renderListingsForType)
          : null}

        {hasAllResources && resourcesCount === 0 ? (
          <NoResults>
            <p>
              <FormattedMessage id="NO_RESOURCES" />
            </p>
          </NoResults>
        ) : null}

        <ButtonBar>{loadMoreButton}</ButtonBar>
      </article>
    );
  }

  private renderListingsForType = (type: string) => (
    <section
      className={cn("ResourceListings", `ResourceListings-${type}`)}
      key={type}
    >
      <h2>
        <FormattedMessage id={type.toUpperCase()} />
      </h2>

      <p>
        <FormattedMessage id={`RESOURCES_${type.toUpperCase()}_DESCRIPTION`} />
      </p>

      <div className="ResourceListings-items">
        {this.props.resources[type].map((resource: IResource) => (
          <ResourceListing
            {...resource}
            key={resource.slug}
            onLoad={this.onListingLoad}
          />
        ))}
      </div>
    </section>
  );

  private onListingLoad = (slug: string) => {
    this.setState({
      loadedListings: {
        ...this.state.loadedListings,
        [slug]: true
      }
    });
  };

  private onLoadMore = () => {
    this.props.fetchMoreResources({});
  };
}

const mapStateToProps = (state: any) => ({
  error: selectors.getResourcesError(state),
  hasAllResources: selectors.getHasAllResources(state),
  isLoading: selectors.getResourcesIsLoading(state),
  resources: selectors.getResourcesByType(state),
  resourcesCount: selectors.getResourcesCount(state)
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      fetchMoreResources: actions.fetchMoreResources.started,
      fetchResources: actions.fetchResources.started
    },
    dispatch
  );

export default injectIntlIntoPage(
  connect<IStoreProps, IDispatchProps>(
    mapStateToProps,
    mapDispatchToProps
  )(ResourcesRoute)
);
