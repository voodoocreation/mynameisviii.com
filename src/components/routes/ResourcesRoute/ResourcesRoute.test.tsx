import { mount, render } from "enzyme";
import { camelizeKeys } from "humps";
import merge from "lodash.merge";
import * as React from "react";
import { Provider } from "react-redux";

import * as selectors from "../../../selectors/root.selectors";
import createStore from "../../../store/root.store";
import ResourcesRoute from "./ResourcesRoute";

import resources from "../../../../server/mocks/resources.json";
import * as dom from "../../../helpers/dom";
import { arrayToAssoc } from "../../../transformers/transformData";

const mockData: any = camelizeKeys(resources);

Object.defineProperty(dom, "isAlmostInViewport", {
  value: jest.fn(() => false)
});

const setup = (fn: any, fromTestStore = {}, fromTestApi?: {}) => {
  const store = createStore(
    merge(
      {
        resources: {
          items: {}
        }
      },
      fromTestStore
    ),
    {},
    fromTestApi
  );

  return {
    actual: fn(
      <Provider store={store}>
        <ResourcesRoute />
      </Provider>
    ),
    store
  };
};

describe("[routes] <ResourcesRoute />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render, {
      resources: {
        hasAllItems: true,
        items: arrayToAssoc(mockData.items, "slug")
      }
    });

    expect(actual).toMatchSnapshot();
  });

  it("renders no results correctly", () => {
    const { actual } = setup(render, {
      resources: {
        hasAllItems: true
      }
    });

    expect(actual.find(".NoResults")).toHaveLength(1);
  });

  it("renders with an error correctly", () => {
    const { actual } = setup(render, {
      resources: {
        error: {
          message: "Server error",
          status: 500
        }
      }
    });

    expect(actual.find(".LoadButton").text()).toBe("Try again");
    expect(actual).toMatchSnapshot();
  });

  it("renders with 'load more' button when hasAllItems=false", () => {
    const { actual } = setup(render, {
      resources: {
        hasAllItems: false
      }
    });

    expect(actual.find(".LoadButton")).toHaveLength(1);
  });

  it("renders without 'load more' button when hasAllItems=true", () => {
    const { actual } = setup(render, {
      resources: {
        hasAllItems: true
      }
    });

    expect(actual.find(".LoadButton")).toHaveLength(0);
  });

  it("loads more listings when 'load more' button is activated", () => {
    const { actual } = setup(
      mount,
      {},
      {
        fetchResources: () => ({
          data: {
            items: [mockData.items[0]]
          },
          ok: true
        })
      }
    );

    expect(actual.find("ResourceListing")).toHaveLength(0);
    actual.find("LoadButton").simulate("click");
    expect(actual.find("ResourceListing")).toHaveLength(1);
  });

  it("updates `loadedListings` state after a listing has loaded", () => {
    const { actual } = setup(mount, {
      resources: {
        items: { "test-1": mockData.items[0] }
      }
    });

    actual.find("ResourceListing img").simulate("load");

    expect(actual.find(".hasLoadedAllListings")).toHaveLength(1);
  });

  describe("getInitialProps()", () => {
    const items = [mockData.items[0]];

    it("fetches items when rendering on the server", async () => {
      const { store } = setup(
        render,
        {},
        {
          fetchResources: () => ({
            data: { items },
            ok: true
          })
        }
      );

      await ResourcesRoute.getInitialProps({
        ctx: {
          isServer: true,
          store
        }
      });

      expect(selectors.getResourcesAsArray(store.getState())).toEqual(items);
    });

    it("fetches items when the store is empty", async () => {
      const { store } = setup(
        render,
        {},
        {
          fetchResources: () => ({
            data: { items },
            ok: true
          })
        }
      );

      await ResourcesRoute.getInitialProps({
        ctx: {
          isServer: false,
          store
        }
      });

      expect(selectors.getResourcesAsArray(store.getState())).toEqual(items);
    });

    it("fetches more items when there are more to fetch", async () => {
      const { store } = setup(
        render,
        {
          resources: {
            items: {
              "test-1": items[0]
            }
          }
        },
        {
          fetchResources: () => ({
            data: {
              items: [{ slug: "test-2" }]
            },
            ok: true
          })
        }
      );

      expect(selectors.getResourcesCount(store.getState())).toBe(1);

      await ResourcesRoute.getInitialProps({
        ctx: {
          isServer: false,
          store
        }
      });

      expect(selectors.getResourcesCount(store.getState())).toBe(2);
    });

    it("doesn't fetch anything when all items are in the store", async () => {
      const { store } = setup(render, {
        resources: {
          hasAllItems: true,
          items: {
            "test-1": items[0]
          }
        }
      });

      await ResourcesRoute.getInitialProps({
        ctx: {
          isServer: false,
          store
        }
      });

      expect(selectors.getResourcesAsArray(store.getState())).toEqual(items);
    });
  });
});
