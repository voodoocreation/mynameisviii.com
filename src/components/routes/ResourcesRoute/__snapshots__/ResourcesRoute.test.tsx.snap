// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <ResourcesRoute /> renders correctly 1`] = `
<article
  class="ResourcesRoute"
>
  <header
    class="PageHeader"
  >
    <h1>
      <span>
        Resources
      </span>
    </h1>
  </header>
  <section
    class="ResourceListings ResourceListings-design"
  >
    <h2>
      <span>
        Design
      </span>
    </h2>
    <p>
      <span>
        Designers can utilise the following files for their promotional material that may mention Viii. Wherever possible, vector artwork is provided and should be used. When using any of the vector resources - their fill should be altered to match the colour scheme of the design.
      </span>
    </p>
    <div
      class="ResourceListings-items"
    >
      <article
        class="ResourceListing"
      >
        <a
          href="https://www.dropbox.com/s/vtht3hip5t6eegu/viii-logo-text-in-circle.svg?dl=0"
          rel="noopener"
          target="_blank"
        >
          <figure
            class="Image ResourceListing-image isLoading"
          >
            <div
              class="Image-wrapper"
            >
              <img
                alt="Text logo (within circle device)"
                src="https://s3.amazonaws.com/mynameisviii-static/resources/viii-logo-text-in-circle.png"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader-animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <div
            class="ResourceListing-details"
          >
            <h3>
              Text logo (within circle device)
            </h3>
            <p>
              The Viii logo variant displaying the branded text within the painted circle device.
            </p>
          </div>
        </a>
      </article>
      <article
        class="ResourceListing"
      >
        <a
          href="https://www.dropbox.com/s/1vbklerzh9yygit/viii-symbol.svg?dl=0"
          rel="noopener"
          target="_blank"
        >
          <figure
            class="Image ResourceListing-image isLoading"
          >
            <div
              class="Image-wrapper"
            >
              <img
                alt="Symbol"
                src="https://s3.amazonaws.com/mynameisviii-static/resources/viii-symbol.png"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader-animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <div
            class="ResourceListing-details"
          >
            <h3>
              Symbol
            </h3>
            <p>
              The Viii symbol, depicting an eye within a triangle within a circle.
            </p>
          </div>
        </a>
      </article>
      <article
        class="ResourceListing"
      >
        <a
          href="https://www.dropbox.com/s/3q09a3r70nzv388/vii-logo-text.svg?dl=0"
          rel="noopener"
          target="_blank"
        >
          <figure
            class="Image ResourceListing-image isLoading"
          >
            <div
              class="Image-wrapper"
            >
              <img
                alt="Text logo (standalone)"
                src="https://s3.amazonaws.com/mynameisviii-static/resources/viii-logo-text.png"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader-animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <div
            class="ResourceListing-details"
          >
            <h3>
              Text logo (standalone)
            </h3>
            <p>
              The Viii logo variant displaying only the branded text.
            </p>
          </div>
        </a>
      </article>
      <article
        class="ResourceListing"
      >
        <a
          href="https://www.dropbox.com/s/yixue32q96yel1y/viii-circle-device.svg?dl=0"
          rel="noopener"
          target="_blank"
        >
          <figure
            class="Image ResourceListing-image isLoading"
          >
            <div
              class="Image-wrapper"
            >
              <img
                alt="Circle device"
                src="https://s3.amazonaws.com/mynameisviii-static/resources/viii-circle-device.png"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader-animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <div
            class="ResourceListing-details"
          >
            <h3>
              Circle device
            </h3>
            <p>
              The painted circle device - most commonly used as part of the 'text within circle' logo variant.
            </p>
          </div>
        </a>
      </article>
      <article
        class="ResourceListing"
      >
        <a
          href="https://www.dropbox.com/s/c599ysb05jp0h2b/sky-day.jpg?dl=0"
          rel="noopener"
          target="_blank"
        >
          <figure
            class="Image ResourceListing-image isLoading"
          >
            <div
              class="Image-wrapper"
            >
              <img
                alt="Sky (day)"
                src="https://s3.amazonaws.com/mynameisviii-static/resources/sky-day.jpg"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader-animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <div
            class="ResourceListing-details"
          >
            <h3>
              Sky (day)
            </h3>
            <p>
              Raster image of the daytime sky, often used in official Viii promotional material.
            </p>
          </div>
        </a>
      </article>
      <article
        class="ResourceListing"
      >
        <a
          href="https://www.dropbox.com/s/mpiagnepdq3nla6/sky-night.jpg?dl=0"
          rel="noopener"
          target="_blank"
        >
          <figure
            class="Image ResourceListing-image isLoading"
          >
            <div
              class="Image-wrapper"
            >
              <img
                alt="Sky (night)"
                src="https://s3.amazonaws.com/mynameisviii-static/resources/sky-night.jpg"
              />
              <span
                class="Loader"
              >
                <div
                  class="Loader-animation"
                >
                  <span />
                  <span />
                  <span />
                </div>
              </span>
            </div>
          </figure>
          <div
            class="ResourceListing-details"
          >
            <h3>
              Sky (night)
            </h3>
            <p>
              Raster image of the night sky, often used in official Viii promotional material.
            </p>
          </div>
        </a>
      </article>
    </div>
  </section>
</article>
`;

exports[`[routes] <ResourcesRoute /> renders with an error correctly 1`] = `
<article
  class="ResourcesRoute hasLoadedAllListings"
>
  <header
    class="PageHeader"
  >
    <h1>
      <span>
        Resources
      </span>
    </h1>
  </header>
  <div
    class="ButtonBar"
  >
    <button
      class="Button LoadButton"
      type="button"
    >
      <span
        class="Button-content"
      >
        <span>
          Try again
        </span>
      </span>
    </button>
  </div>
</article>
`;
